### PR TITLE
adding a release github action

### PR DESCRIPTION
**Pull request by andykruth-ii**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 name: Build
 
-on: [push, pull_request]
+on:
+  push:
+    branches-ignore:
+      - master
```
Comment (techabstraction) [inline .github/workflows/build.yml:6]: if we are skipping tests on the release pipeline then we should run them here for master builds?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          gpg-passphrase: GPG_PASSPHRASE # env variable for GPG private key passphrase
+          settings-path: ${{ github.workspace }}
+      - name: Cache local Maven repository
+        uses: actions/cache@v2.1.6
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Release artifacts
+        run: mvn -Dresume=false release:prepare release:perform -Djib.skip -DskipTests -Darguments="-s $GITHUB_WORKSPACE/settings.xml"
```
Comment (techabstraction) [inline .github/workflows/release.yml:36]: set release version and development version here?

File: .github/workflows/build.yml (hunk 1 above)
Comment (andykruth-ii) [reply .github/workflows/build.yml]: reverting back to always doing a build.

File: .github/workflows/release.yml (hunk 2 above)
Comment (andykruth-ii) [reply .github/workflows/release.yml]: done

**Review by jamespfaulkner — APPROVED** [review]

**Pull request merged**